src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -87,13 +87,35 @@ public static void ToCharsBuffer(byte value, Span<char> buffer, int startingInde
         }
 
 #if SYSTEM_PRIVATE_CORELIB
+        // Converts Vector128<byte> into 2xVector128<byte> ASCII Hex representation
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static (Vector128<byte>, Vector128<byte>) AsciiToHexVector128(Vector128<byte> src, Vector128<byte> hexMap)
+        {
+            Debug.Assert(Ssse3.IsSupported || AdvSimd.Arm64.IsSupported);
+            // The algorithm is simple: a single srcVec (contains the whole 16b Guid) is converted
+            // into nibbles and then, via hexMap, converted into a HEX representation via
+            // Shuffle(nibbles, srcVec). ASCII is then expanded to UTF-16.
+            Vector128<byte> nibbles = Vector128.ShiftRightLogical(src.AsUInt64(), 4).AsByte();
```

Comment:
Nit: is `nibbles` really the right name for this?  This is just the source data but shifted such that it's missing 8 of the bits.